src/util/checkoutUtils.js

```diff
@@ -120,3 +121,68 @@ export function removeCredit(apollo, creditType) {
 		}
 	});
 }
+
+/**
+ * Poll the checkoutStatus endpoint until the checkout is complete
+ * Note: We only operate on the COMPLETED or results with errors
+ *
+ * Possible status values:
+ * - BASKET_MANIFEST
+ * - BASKET_VALID
+ * - CHECKOUT_FAILED
+ * - CHECKOUT_RECORDED
+ * - CHECKOUT_ROLLED_BACK
+ * - COMPLETED
+ * - CREDIT_ADDED
+ * - DEPOSIT_RECORDED
+ * - FAILED
+ * - MANIFEST_FAILED
+ * - RECORD_CHECKOUT_FAILED
+ * - REQUEST_RECEIVED
+ * - RESERVATIONS_COMPLETED
+ * - STARTED
+ * - TRANSIENT_PAYMENT_METHOD_CHARGED
+ *
+ * @param {Object} apollo Apollo Client instance
+ * @param {Number} transactionId
+ * @param {Number} interval How often to poll
+ * @param {Number} timeout How long to allow polling to continue
+ * @returns {Promise}
+ */
+export async function pollForCheckoutStatus(
+	apollo = null,
+	transactionSagaId = 0,
+	interval = 1000,
+	timeout = 60000
```

Comment:
The existing checking process already times out after 60 seconds, so it seems like this will have the same problem. What if we went with 5 minutes? That's the default connection timeout for most browsers.